android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/network/AsyncLinkedNetworkManagerTest.java

```diff
@@ -73,6 +78,48 @@ public void testConstructor() {
         assertNotNull(linkedNetworkManager);
     }
 
+    @Test
+    public void testMimeTypeFinder() {
+        AndroidMimeTypeFinder androidMimeTypeFinder = mock(AndroidMimeTypeFinder.class);
+        androidMimeTypeFinder.getMimeType(new FileMetaData(), new InputStream() {
+            @Override
+            public int read() throws IOException {
+                return -1;
+            }
+        });
+        verify(androidMimeTypeFinder, times(1)).getMimeType((FileMetaData) any(), (InputStream) any());
+        verify(androidMimeTypeFinder, never()).getMimeType((FileMetaData) any());
+    }
+
+    @Test
+    public void testMimeTypeFinderInputStream() {
+        AndroidMimeTypeFinder androidMimeTypeFinder = new AndroidMimeTypeFinder();
+        FileMetaData fileMetaData = new FileMetaData();
+        androidMimeTypeFinder.getMimeType(fileMetaData, new InputStream() {
+            @Override
+            public int read() throws IOException {
+                return -1;
+            }
+        });
+        assertEquals(fileMetaData.getMimetype(), "application/octet-stream");
+    }
+
+    @Test
+    public void testMimeTypeFinderNullFile() {
+        AndroidMimeTypeFinder androidMimeTypeFinder = new AndroidMimeTypeFinder();
+        FileMetaData fileMetaData = new FileMetaData();
+        androidMimeTypeFinder.getMimeType(fileMetaData, (File) null);
+        assertEquals(fileMetaData.getMimetype(), "application/octet-stream");
+    }
+
+    @Test
+    public void testMimeTypeFinderNullMimetype() {
+        AndroidMimeTypeFinder androidMimeTypeFinder = new AndroidMimeTypeFinder();
+        FileMetaData fileMetaData = new FileMetaData();
+        androidMimeTypeFinder.getMimeType(fileMetaData);
+        assertEquals(fileMetaData.getMimetype(), "application/octet-stream");
```

Comment:
How is this testing for a null Mimetype?

Reply:
The method has the following code:
`@Override`
   ` public void getMimeType(FileMetaData metaData) {`
     `   if (metaData.getMimetype() != null && metaData.getMimetype().length() > 0) {`
         `   return;`
       ` }`
     `   String mimetype = null;`

        if (metaData.getFileName() != null) {
            int dotIndex = metaData.getFileName().lastIndexOf("");

            if (dotIndex > 0 && dotIndex + 1 < metaData.getFileName().length()) {
                mimetype = MimeTypeMap.getSingleton().getMimeTypeFromExtension(metaData.getFileName().substring(dotIndex + 1, metaData.getFileName().length()));
            }
        }

        if (mimetype == null) {
            mimetype = "application/octet-stream";
        }
        metaData.setMimetype(mimetype);
    }`
If we put `metaData` with `metaData.getMimetype() == null` and `metaData.getFileName() == null` in the method, we will have `fileMetaData.getMimetype().equals( "application/octet-stream")`